packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -264,11 +293,26 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
       prev: this.tip,
       id: this.id.cid,
     }
+    // Null check is necessary to avoid `undefined` value that can't be encoded with IPLD
+    if (shouldIndex != null) {
+      rawCommit.header = {
+        shouldIndex: shouldIndex,
+      }
+    }
     const commit = await signer.createDagJWS(rawCommit)
-    const updated = await this.api.applyCommit(this.id, commit, opts)
+    const updated = await this.api.applyCommit(this.id, commit, options)
     this.state$.next(updated.state)
   }
 
+  /**
+   * Set the index metadata field for the stream
+   * @param shouldIndex - Whether the stream should be indexed or not
+   * @param opts - Additional options
+   */
+  shouldIndex(shouldIndex: boolean, opts: CommonUpdateOpts = {}): Promise<void> {
+    return this.replace(this.content, { ...opts, shouldIndex: shouldIndex })
```

Comment:
Wouldn't it be more efficient to use `.patch()` directly to bypass the JSON-patch logic?
```suggestion
    return this.patch([], { ...opts, shouldIndex: shouldIndex })
```

Reply:
True.